lib/src/lib.rs

```diff
@@ -83,3 +89,18 @@ impl Files {
         }
     }
 }
+
+/// Iterates over the file names available for `Files` object.
+pub struct FileNames<'a> {
+    /// Our internal iterator.  We wrap this in a nice struct so our
+    /// caller doesn't need to know the details.
```

Comment:
Documentation for non-public items is stripped by default, so you could save some slashes here. Not that you needed to.